### ISSUE TITLE
Decide on a long-term plan to create an official location for community discussions

### ISSUE THREAD

**Issue by philsward** [backdrop-ops/backdropcms.org#778]
### Introduction
After some discussion in Backdrop Live, it was realized that the community doesn’t have an official location to create and discuss issues that are about the community. Currently, many of these discussions live within the /backdrop-ops/backdropcms.org repo which is arguably a really bad location for community discussion because that repo should preferably only deal with the code and content of the backdropcms.org website.

The idea was presented in a [Zulip topic](https://backdrop.zulipchat.com/#narrow/stream/218635-Backdrop/topic/New.20Github.20Group.20For.20Community.20Discussion) to create repos for key areas of discussion that will or may require the ability to create documents for policies etc and track the progress of how the content in those documents came to be. Through this discussion, It was discovered that there are a few main ideas on how to approach community discussion and we need to come up with a game plan on the best approach for the long term.

Regardless of the medium or the physical structure of these discussions, it is believed that a repo or group of repos should still be used to create the official policies, guidelines and documents around these discussions when relevant. For example, code of conduct needs a Standard Operating Procedure (SOP) to describe policies. Outreach on the other hand, may not need this SOP. Either way, it makes sense to have a place where policies can be treated as trackable code.

### The Ideas

**Idea 1)** Create a single “community” repo inside github.com/backdrop-ops

This idea is to have a single repo inside the /backdrop-ops organization and create labels for each of the different ongoing topics. For anyone not familiar, /backdrop-ops is the “Backdrop Operations” Github organization that was originally designated to house “everything else” that isn’t related to core or contrib.

The primary reason for using this single repo to house all community discussion, basically boils down to the least amount of overhead required to get it going and maintain it with the limited amount of manpower currently available to the community.

The current name for this repo has been suggested to be “community” so it would live in github.com/backdrop-ops/community

The merits to using a single repo approach are less overhead thus less manpower required to maintain the repo. It puts all topics of discussion into one single repo, separated out by labels.

The pitfalls are a lack of true organization and separation of topics within the community. There is also the problem of trying to weed through code repos to find non-code repos that live within backdrop-ops. At the time of this writing, there are approximately 17 repos, 3 of which are not code related and 2 of those are hidden from the public. If documents are created to outline policies or guidelines, the front page of this single repo will have a list of multiple documents for each topic and some of those topics may have 2 or more documents making them potentially difficult to sift through as the document list grows. While labels have been suggested as a means to segregate topics, everyone must be given permission to create labels which creates additional maintenance overhead to grant every participating member the ability to label their own discussion topic (maybe Github allows the public to create labels without being inside the organization and I am unaware). In addition, once you are a member of an organization you end up getting notifications of every single repo within that organization. Overall it’s a great approach for short-term but doesn’t appear to scale well long term.

**Idea 2)** Create a new organization and create repos for each

As a disclaimer, this is the proposal I have made so it might sound a bit biased or more inclusive.

The main idea is to create a whole new organization and create a new repo for each and every top level topic. Some of the topics currently living in -ops should be moved over as well. The following are examples of possible (and existing) top level ongoing topics that could be created as separate repos:

- Outreach
- Policies
- Philosophy
- Roadmap
- Funding
- Infrastructure-Overhead
- Events
- Initatives
- Branding
- License-Copyright
- Conduct
- Leadership (private)
- PMC (private)
- General (catch-all)

The main goal to having a separate organization with repos for topics is to be able to point members to that specific github organization and explain that all you will find in that organization are community topics for discussion. No code, nothing else.

This idea believes that ops should be relegated to “everything else that is code” and have the new organization group relegated to “everything else that is not code”.

The merits to this idea are that the overall organization of topics and containing all discussion that is pertinent to that particular topic. This idea also allows people to star and watch the topics that interest them. In contrast, if we go with a single community repo inside of ops, anyone watching that repo will see every issue about every topic whether it interests them or not. With this approach, we don’t have to worry about the general community needing access to create labels. By separating discussions out into topics, we don’t need a large set of labels, only a handful that need to draw attention to a call-to-action on that issue making it very manageable by a few people without giving unnecessary access to a plethora of members.

The main pitfall is manpower and overhead required to create and potentially maintain this new area. Realistically, how much overhead is required to maintain a new org vs a single repo? Setup is obviously more overhead, but is the maintenance any higher or would it be about the same as a single repo?

**Idea 3)** Create a place for discussion using Backdrop

Similar to the forums (which has been suggested) this idea would encompass discussion directly using either the primary .org site or on a subdomain.

The merits to this idea is that it creates a familiar place to house discussion, new members don’t have to specifically create Github accounts to participate, the organization of the discussions can be customized to our liking such as adding a polling and voting mechanism that is tied directly into the issue or creating permissions that are specific to the needs of the user or discussion. These are a few of many benefits to doing it in-house.

The pitfalls are the high cost of overhead to not only create this area, but also maintain it. Short term this will be a problem but long term it shouldn’t be as much of an issue as the community grows and more folks can help out. Since this would be hosted internally, there is the risk of the site going down and ongoing updates to keep it secure etc. (refer back to maintenance overhead)

### Additional Thoughts

It was brought up that one problem to creating a new organization is that it goes against the ongoing discussions of narrowing the places where discussion can take place. For example there is Zulip, Github, forums, social media, zoom, etc. This is a valid point that might need to be addressed however from what I have gathered, the idea is to condense the number of mediums, not the rooms within those mediums. Maybe I’m missing something to that discussion but I don’t believe that adding a new organization to Github will go entirely against this discussion.

### Where do we go from here

In the short term we need a place for discussion that isn’t housed within a code oriented repo like backdropcms.org. The plan I propose is a new Github organization created with repos that house the separate topics, that way we can have each repo contain the specific policies and documents for those topics. This will be a short term solution with the ultimate goal of bringing these discussions in-house within the b.org umbrella, but still using this Github organization to hold and track the official documentation.

I personally believe we need to think about this for the long term of 5+ years out and minimizing long-term work later by creating a solid foundation now. I’m fine with a single repo within backdrop-ops as long as we all agree on what the “next step” is and agree it needs to be fixed relatively quickly. But I am of the very strong opinion that a single repo will not work efficiently as a long-term solution and will need to be changed in the future anyway. I would like to mitigate that future work now.

I would like us to spend the next few weeks gathering some votes on this idea so we can move forward with a game-plan.

If you want to share some thoughts or experiences that’s fine, but let’s try to move on this discussion within the next 30 - 45 days so we can quickly open up a place to discussion things about the community, regardless of where we decide it should be housed. There are formal community discussions that need to be raised and they are being blocked by this issue.

**Comment by philsward** [comment]
**Note:** there is ongoing discussion in Zulip so if you would like additional context and up-to-date discussion, peek in the topic for additional context: https://backdrop.zulipchat.com/login/#narrow/stream/218635-Backdrop/topic/New.20Github.20Group.20For.20Community.20Discussion

**Comment by stpaultim** [comment]
@philsward Thank you for creating this ticket and raising this issue and providing all the details. You are not the first to suggest something along these lines and I think that you are trying to solve a real problem. There are reasons why this was not done in the past, but as our community grows the situations change. 

In the past, we "officially" used the backdropcms.org issue queue for these discussions, specifically because we didn't want to create new spaces for discussion that would confuse and discourage new participants. Also, we had limited number of community issues and a limited number of people that were interested in participating in those discussions. 

It's not clear to me that our previous policy was the best long term policy, but it seemed to work for us at the time, at least better than any alternative we could come up with. I know that one of my biggest frustrations as a new contributor was trying to remember all the different discussions venues and which one was the the correct one for each topic. I remember times in which I knew there was a specific forum for a specific topic, but it would take me 15 minutes to find that forum (among all the other forums and repositories). This page was created about 2.5 years ago to help solve that problem.
https://backdropcms.org/resources/backdrop-cms-github

The forum was created prior to my involvement in the project, because (as I understand it) we wanted a place for Backdrop users to post questions about the project that was not as intimidating as Github. For people who don't write code or have experience with Github, it can be an intimidating tool and we wanted someplace that was more familiar and usable for users to ask questions. 

Since the creation of the forum, we have periodically discussed moving some "community" related topics to the forum. I think that the two biggest reasons we have avoided that is:

1) Many of the active contributors in the community don't really need or use the forum. They are already in Github and are more likely to participate in discussions in Github. 
2) The forum is nice for it's current usage, but it doesn't have all the features and tools that Github does. Also, sometimes community issues become software issues and it saves a step if those discussions start in Github. 

I support the idea of a dedicated forum/queue for community issues. I am not sure that we need multiple community issue queues. I think that have multiple queues for different projects might further confuse users that are already confused by our maze of github repos and issue queues. BUT, I confess to frequently wishing we had a distinct queue for a specific initiative or specific project. I worry that while this would be helpful for me, it would confuse and intimidate new users. 

While I support the idea of a new forum/queue for community issues, I would feel much better if we first came up with a nice infographic or tool that helps new users navigate all the forums/queues that we already have.

Would love to hear from some folks who have come to the community more recently about their own experiences and frustrations with all the various communications channels we have and how we can make it easier.

**Comment by BWPanda** [comment]
> I would feel much better if we first came up with a nice infographic or tool that helps new users navigate all the forums/queues that we already have.

If we need an infographic/tool to help users navigate and find the right place to post something, then we already have too many options... IMO.

**Comment by stpaultim** [comment]
> If we need an infographic/tool to help users navigate and find the right place to post something, then we already have too many options... IMO.

@BWPanda - This is an easy statement to make and could be true. But, I'm lacking any ideas for a solution that doesn't involve some kind of infographic or tool. In fact, one could say that if we need a special page like: backdropcms.org/resources/backdrop-cms-github, then we already have too many places. 

Yet, I'm glad we have that page, because I still use it on a frequent basis and I don't know what the alternative is - OTHER than an infographic which would be an improvement on this page. 

So, I agree that my preferred solution would be to create a structure that does not require an infographic or a page like backdropcms.org/resources/backdrop-cms-github, but in the last 3 years that I've been thinking about this problem - I have yet to hear any ideas that meet all our various needs and don't require this kind of documentation. 

What ideas do you have to organize our forums and queues that would be easy for a new user to navigate without the aid of an infographic or tool?

**Comment by BWPanda** [comment]
IMO, the best solution is to provide a place where users can post things without needing to worry about if its the right place, etc. Once they've posted something, the rest of the community can then organise it appropriately and as necessary.

For example: users can post in Zulip with whatever topic name they like. Mostly it just stays that way, which is fine, but we can easily rename the topic (or move the comments) if necessary. Or, users can post issues to the issue queue with no label, but then we can label it appropriately.

In both cases, the burden of organisation is not on users but on the rest of us (mainly the moderators/core group of users of each medium). And that's how I think it should be.

To bring it back to this particular issue, if we create multiple repos/issue queues, there's no way a user can post 'generically' - they have to decide first *where* their post should go. If, however, we have a single repo/issue queue, then users can post there without needing to decide anything, and then labels can be added after the fact once we've decided what sort of content it is.

I'm not against infographics/documentation if it helps people with the problem of multiple places to post, but I'd much rather fix the problem so that additional documentation isn't necessary.

**Comment by docwilmot** [comment]
Precis version: IMHO we already have way too many locations and I am concerned that we would do more harm by introducing a bunch more rather than optimizing the ones we've already got.

**Comment by philsward** [comment]
I think the biggest disconnect between having too many areas and it being confusing is that there is zero organization to even display those areas. The mediums are fairly straightforward, but where on b.org does it say what ops is, where it is and what goes in it?

There is too much "hiding" of locations through text with links instead of spelling out exactly where to look. Text links point to specific github repos instead of the github organization and there is little to no text defining and explaining the organization or the repos themselves.

As a new user, it was (and still is) daunting to figure out what's what and where stuff lives because nothing is cut and dry. The people who will use these mediums aren't your average computer novice but it's agreed that these areas shouldn't be intimidating either.

In my mind, putting everything inside a single repo puts more strain on a few members of the community perpetually. Saying the members aren't smart enough to figure out where to put things, is a bit of an insult unless the locations are so cryptic that nobody can figure it out anyway which is what we have now. Let the members choose where to file an issue and give them the full scope of the github organization instead of walling it off and trying to hold their hand on where to go. Single repo means 100% maintenance by "someone" to manage labels. Separate repos means a lot less long term maintenance overhead if only a few issues require labels. I've also suggested a repo specifically for topics that aren't otherwise specified, as the catch-all if people can't decide where to post. People also have the option to simply ask in the forums or Zulip if they are unsure which repo to post in.

I have raised the proposal of a new org and separate repos because I have personally found the disorganization and lack of information about discussion to be far more confusing than having more areas for people to discuss things. No offense intended to anyone who has spent time on creating the content.

The next issue I will raise is to rework how the b.org site deals with locations of discussion and code. Until then, one thing at a time.

**Comment by stpaultim** [comment]
>

**Comment by yorkshire-pudding** [comment]
As a newish member, I've hardly used the forum.  I first came to join the community after having bug with bootstrap_lite and wanting to send a fix.  I could put a comment on the github issue, which I did, but didn't know how long it would take before someone saw it.  I quickly found Zulip and it is pretty easy to use on phone and in browser.  I could post a thread with a link to github and ask for advice from the wider community.  Within a few hours, I got the advice I needed to do a pull request.  That was good.  I like the idea of Idea 1 - a single repo with labels - keep it simple rather than split.  This mirrors how Zulip works.  To draw comparisons from other data worlds, I so often see spreadsheets with data needlessly spread out over multiple worksheets, making it much harder to analyse; likewise, its easy to filter to a single stream in Zulip and I think it would be in a repo with labels.  I see a single community repo having wider discussion and contribution to each issue than narrow repos.

**Comment by BWPanda** [comment]
That's a good point @yorkshire-pudding, and one that I made re. moving all documentation into one website - you only need to search one place to find what you're looking for, rather than multiple places.

**Comment by philsward** [comment]
Doing some playing around with github, it's actually a really bad place for general discussions because of the lack of public permissions.

The new "Discussions" would be excellent for this issue, however it appears that only members of an organization are allowed to create new discussions... (It's in beta, so maybe this will change down the road). It's possible to automate adding people as members but that will require someone to create the script for it. It would need to be as simple as a person filling out a form on b.org by adding their email and it would add them as a member into the organization on github.

It also appears that only members are allowed to add labels to issues so I have to ask: who wants to volunteer for adding a label to every single issue in order to keep it organized? If the community grows to 100,000 members and there are on average 1000 discussion issues created daily, how many people will we need to manage a single repos label for topics? This overhead isn't necessary with a separate org with individual topic repos, but that route has its own logistical hurdles to overcome as mentioned above.

I'm working on a mock-up organization based on how I would envision it to work as a separate organization with individual repos used for specific topics of discussion. I'll share when it is closer to being usable.

What is nice is any issue or discussion can be moved to any other repo within the organization. Meaning, if someone posts in the wrong repo (the big fear), it's easy to move it to the right one.

**Comment by stpaultim** [comment]
> Meaning, if someone posts in the wrong repo (the big fear)

In my opinion, this is not the biggest fear. The biggest fear is that people won't post at all because they don't understand the system, don't know where to post, or can't easily find the "right" repo. Personally, I would rather people post in the wrong place than not post at all, so this (posting in the wrong place) is NOT a big fear of mine.

**Comment by philsward** [comment]
@stpaultim and like I've said before, explain to people where to post and it won't be a problem.

As a personal experience example, I had zero information to go on when I came to the community and it said I could chat live on Zulip. Exciting! I posted to the wrong stream and not knowing the atmosphere, felt like I got my hands slapped for not knowing what I was doing. It was embarrassing through my own lack of knowledge and could have been avoided with a bit more context going in.

Having a single generic repo is still going to create the same sense of "Is this the right place to ask my question?" So it doesn't really matter if there is one repo for all topics or a handful of repos for individual "key" topics.

Keep in mind, this area for discussion is supposed to be about creating game-plans, policies and structure around the community. This is a place for members who want to be involved with the direction and growth of Backdrop. If members truly are passionate, they'll figure it out one way or another despite their insecurities but we can do a much better job up front by providing the right information to begin with.

**Comment by philsward** [comment]
@stpaultim curious, do you know of anyone personally who has not posted to any of the mediums because they were too scared they would post in the wrong place?

**Comment by philsward** [comment]
Link to the discussions sandbox:

https://github.com/backdrop-talk

Note: I have no idea how permissions will play out... Organizations are intended for people to be invited as opposed to the general public. If you want access to check out the "discussions" area, let me know and I'll add you as a member to play around.

I encourage you to also compare this to: https://github.com/backdrop-ops which is where the single repo idea would be housed.

**Comment by jenlampton** [comment]
> @stpaultim curious, do you know of anyone personally who has not posted to any of the mediums because they were too scared they would post in the wrong place?

It's not exactly the same thing, but I don't often use Zulip because it takes me so long to figure out where the heck to post something there, that it ends up not being worth the effort of posting it. Fear is not the problem, it's having a forced choice that's the blocker. The more forced choices we give people, the less activity we're going to see, as only the really motivated people will learn the system. 

As far as "knowing anyone personally" - we may never get to know the people who don't ever make that first post!

I agree with others on this thread that we already have too many options, and I would prefer to avoid adding anything else so that we don't make that initial decision on where to post something any harder than it already is. I agree with @BWPanda that the number of options is a problem, an that we should aim to improve it. But I also agree with @stpaultim that until we get there, a graphic for that initial decision could be helpful :) 

> Decide on a long-term plan to create an official location for community discussions

I believe the "official location for community discussions" is still the Forum. 

Since the discussions sandbox link is currently a 404, and hasn't been any movement on this issue for more than a year, I don't think there will be any objections to closing this issue.
